src/AspNet.Security.OAuth.Streamlabs/StreamlabsAuthenticationOptions.cs

```diff
@@ -0,0 +1,77 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+using System;
+using System.Security.Claims;
+using System.Text.Json;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using static AspNet.Security.OAuth.Streamlabs.StreamlabsAuthenticationConstants;
+
+namespace AspNet.Security.OAuth.Streamlabs
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="StreamlabsAuthenticationHandler"/>.
+    /// </summary>
+    public class StreamlabsAuthenticationOptions : OAuthOptions
+    {
+        public StreamlabsAuthenticationOptions()
+        {
+            ClaimsIssuer = StreamlabsAuthenticationDefaults.Issuer;
+            CallbackPath = StreamlabsAuthenticationDefaults.CallbackPath;
+
+            AuthorizationEndpoint = StreamlabsAuthenticationDefaults.AuthorizationEndPoint;
+            TokenEndpoint = StreamlabsAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = StreamlabsAuthenticationDefaults.UserInformationEndpoint;
+
+            ClaimActions.MapCustomJson(ClaimTypes.NameIdentifier, user => GetData(user, "streamlabs.id"));
+            ClaimActions.MapCustomJson(ClaimTypes.Name, user => GetData(user, "streamlabs.username"));
+            ClaimActions.MapCustomJson(Claims.DisplayName, user => GetData(user, "streamlabs.display_name"));
+            ClaimActions.MapCustomJson(Claims.FacebookId, user => GetData(user, "facebook.id"));
+            ClaimActions.MapCustomJson(Claims.FacebookName, user => GetData(user, "facebook.name"));
+            ClaimActions.MapCustomJson(Claims.Primary, user => GetData(user, "streamlabs.primary"));
+            ClaimActions.MapCustomJson(Claims.Thumbnail, user => GetData(user, "streamlabs.thumbnail"));
+            ClaimActions.MapCustomJson(Claims.TwitchDisplayName, user => GetData(user, "twitch.display_name"));
+            ClaimActions.MapCustomJson(Claims.TwitchIconUrl, user => GetData(user, "twitch.icon_url"));
+            ClaimActions.MapCustomJson(Claims.TwitchId, user => GetData(user, "twitch.id"));
+            ClaimActions.MapCustomJson(Claims.TwitchName, user => GetData(user, "twitch.name"));
+            ClaimActions.MapCustomJson(Claims.YoutubeId, user => GetData(user, "youtube.id"));
+            ClaimActions.MapCustomJson(Claims.YoutubeTitle, user => GetData(user, "youtube.title"));
```

Comment:
Looks like all of these can be mapped with `MapJsonSubKey()` and then `GetData()` can be deleted as it becomes redundant.

Here's an example from the Patreon provider:

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/0d9f3f21a7561d258b37ceece293df6ed57fc21b/src/AspNet.Security.OAuth.Patreon/PatreonAuthenticationOptions.cs#L32-L37

Reply:
This is much better. Thanks for the suggestion. https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/pull/553/commits/3137c89a32a8d232b76e6d2b69058c025ad82033